src/content-distribution/incoming-post/index.js

```diff
@@ -0,0 +1,135 @@
+
+function IncomingPost() {
+
+	const { createNotice } = useDispatch( 'core/notices' );
+	const { lockPostAutosaving, unlockPostAutosaving } = useDispatch( 'core/editor' );
+	const { openGeneralSidebar } = useDispatch( 'core/edit-post' );
+	const [ isLinkedToggling, setIsLinkedToggling ] = useState( false );
+	const [ isUnLinked, setIsUnLinked ] = useState( false );
+
+	const { postId } = useSelect( select => {
+		const {
+			getCurrentPostId,
+		} = select( 'core/editor' );
+		return {
+			postId: getCurrentPostId(),
+		};
+	} );
+
+	useEffect( () => {
+		setIsUnLinked( unlinked );
+	}, [ unlinked ] );
+
+	useEffect( () => {
+		setTimeout( () => {
+			openGeneralSidebar(
+				'newspack-network-incoming-post/newspack-network-distribute-panel'
+			);
+		}, 10 ); // TODO. There must be a better way
+	}, [] );
+
+	useEffect( () => {
+		createNotice(
+			'warning',
+			isUnLinked
+				? sprintf( __( 'Originally distributed from %s.', 'newspack-network' ), originalUrl )
+				: sprintf( __( 'Distributed from %s.', 'newspack-network' ), originalUrl ),
+
+			{
+				id: 'newspack-network-incoming-post-notice',
+			}
+		);
+
+		const lockName = 'distributed-incoming-post-lock';
+		if ( isUnLinked ) {
+			unlockPostAutosaving( lockName );
+		} else {
+			lockPostAutosaving( lockName );
+		}
+		// Toggle the CSS overlay.
+		document.querySelector( '#editor' )?.classList.toggle( 'newspack-network-incoming-post-linked', ! isUnLinked );
+
+	}, [ isUnLinked ] );
+
+	const toggleLinked = () => {
+		setIsLinkedToggling( true );
+		apiFetch( {
+			path: `newspack-network/v1/content-distribution/unlink/${ postId }`,
+			method: 'POST',
+			data: {
+				unlinked: !isUnLinked,
+			},
+		} ).then( data => {
+			setIsUnLinked( data.unlinked );
+			createNotice( 'info', __( sprintf( 'Post has been %s.', isUnLinked ? 'unlinked' : 'relinked' ), 'newspack-network' ), {
+				type: 'snackbar',
+				isDismissible: true,
+			} );
+		} ).catch( error => {
+			createNotice( 'error', error.message );
+		} ).finally( () => {
+			setIsLinkedToggling( false );
+		} );
+	}
+
+	return (
+		<DistributePanel
+			header={
+				isUnLinked ? __(
+						'This post has been unlinked from the origin post. Edits to the origin post will not update this version.',
+						'newspack-network'
+					)
+					: __(
+						'This post is linked to the origin post. Edits to the origin post will update this version.',
+						'newspack-network'
+					)
+			}
+			buttons={
+				<>
+					<Button
+						variant="secondary"
+						target="_blank"
+						href={ originalUrl }
+					>
+						{ __( 'View origin post', 'newspack-network' ) }
+					</Button>
+					<Button
+						variant={ isUnLinked ? 'primary' : 'secondary' }
+						isDestructive={ !isUnLinked }
+						disabled={ isLinkedToggling }
+						onClick={ () => {
+							toggleLinked();
+						} }
+					>
+						{ !isUnLinked ? __( 'Unlink from origin post', 'newspack-network' ) : __( 'Relink to origin post', 'newspack-network' ) }
+					</Button>
```

Comment:
It'd be nice to have the `isBusy` state here when it's updating, with the label updated to "Linking..." or "Unlinking...", [like we have for distributing](https://github.com/Automattic/newspack-network/blob/2d676c2c0da66bdac0d6c9b1f342d5c51d3dc657/src/content-distribution/distribute/index.js#L217-L234).